### PR TITLE
Sockets work with multiple clients

### PR DESCRIPTION
**Pull request by Serwlashev**
We have a few conflicts. Please, leave my solution, it works. Now multiple clients can connect to the server and exchange data with it. Also solved some problems with formatting, alignment. Made algorithm receiving data more sophisticated. Now we add a header to the message before sending data, then while receiving we read a number of bytes and create a buffer according to it.  It works faster than receiving data byte by byte.
Problems - clients do not disconnect from the server so cycle still works after they leave the server.

### CODE REVIEW

File: Sockets/BackEndMonitoringSockets/WindowsUtils.cpp [hunk 1]
```diff
@@ -0,0 +1,47 @@
+#include "stdafx.h"
+#include ""
```
Comment (AntonVoytov) [inline Sockets/BackEndMonitoringSockets/WindowsUtils.cpp:2]: What is this include?

File: Sockets/BackEndMonitoringSockets/include/CAcceptor.h [hunk 2]
```diff
@@ -1,23 +1,29 @@
...
+class CServiceHandler;
+
 // Class for accepting connection by the server
 class CAcceptor
 {
 public:
-	CAcceptor(const int port, const std::string& ip_address, 
-		std::shared_ptr<CLogger> logger);
-	int GetConnectedHandle();
+	CAcceptor(const int port, const std::string& ip_address, bool is_blocked);
```
Comment (AntonVoytov) [inline Sockets/BackEndMonitoringSockets/include/CAcceptor.h:11]: CAcceptor class should have next public methods:
CAcceptor(const SAcceptorParams &); // timeout and is_blocked
bool Initialize(const std::string& ip_address, const int listener_port, const int connections); //last parameter for listen function
bool Accept(CSocketWrapper *& p_connection_socket);
bool IsTimeOutWithoutConnections(); // returns true if ::select returns timeout needed for AcceptorWrapper 

All other functions should be private or protected.

File: Sockets/BackEndMonitoringSockets/include/CAcceptorWrapper.h [hunk 3]
```diff
@@ -1,30 +1,39 @@
...
+class CSocketWrapper;
 
 // This class works with server 
 class CAcceptorWrapper
 {
 public:
-	CAcceptorWrapper(int port, const std::string& ip_address, 
-		CEvent& event, std::shared_ptr<CThreadPool> m_pool,
-		std::shared_ptr<CLogger> logger);
+	CAcceptorWrapper(int port, const std::string& ip_address, CEvent& event,
```
Comment (AntonVoytov) [inline Sockets/BackEndMonitoringSockets/include/CAcceptorWrapper.h:15]: CAcceptorWrapper should have the next public methods:
CAcceptorWrapper(const SAcceptorWrapParams &);
bool Inittialize();// InitializeWSA , Create CAcceptor object and initialize it.
bool Execute():// this method should run in other thread and checking global stop event, create a new connection using CAcceptor and enque it in the thread pool.
Maybe setter for thread poll will be need.

**Review by AntonVoytov — COMMENTED** [review]
Please review my comments and if you will have enough time to rework please do it.
Also, please add other guys to review this code.

File: Sockets/BackEndMonitoringSockets/WindowsUtils.cpp (hunk 1 above)
Comment (Serwlashev) [reply Sockets/BackEndMonitoringSockets/WindowsUtils.cpp:2]: It is really strange. I see #include "PlatformUtils.h". I don't know why git doesn't show it.

**Pull request merged**